app/subscriber/src/features/manage-folder/ManageFolder.tsx

```diff
@@ -0,0 +1,177 @@
+  const { id } = useParams();
+  const [, { getFolder, updateFolder }] = useFolders();
+  const navigate = useNavigate();
+  const [, { findContent }] = useContent();
+  const [folder, setFolder] = React.useState<IFolderModel>();
+  const [items, setItems] = React.useState<any>([]);
+  const [sortedFlag, setSortedFlag] = React.useState(false);
+  const [selected, setSelected] = React.useState<IContentModel[]>([]);
+
+  /** TODO: Folder content only contains contentId and sortOrder so we have to make an additional call based off of the contentIds to get the headline/summary etc..
+   * assuming we want this to differ eventually.
+   */
+  React.useEffect(() => {
+    getFolder(Number(id)).then((data) => {
+      // have to use temp not folder state as folder state will not update right away
+      const temp = data;
+      setFolder(data);
+      findContent({
+        contentTypes: [],
+        contentIds: data.content.map((c) => c.contentId),
+      }).then((data) => {
+        const tempSort = data.items.map((item) => ({
+          ...item,
+          sortOrder: temp.content.find((c) => c.contentId === item.id)?.sortOrder ?? 0,
+        }));
+        tempSort.sort((a, b) => a.sortOrder - b.sortOrder);
+        setItems(tempSort);
+      });
+    });
+    // Only on initialize, or when sort order changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [sortedFlag]);
+
+  /** function that runs after a user drops an item in the list */
+  const handleDrop = (droppedItem: any) => {
+    if (!droppedItem.destination) {
+      return;
+    }
+    var updatedList = [...items];
+    // Remove dragged item
+    const [reorderedItem] = updatedList.splice(droppedItem.source.index, 1);
+    // Add dropped item
+    updatedList.splice(droppedItem.destination.index, 0, reorderedItem);
+    // Update State
+    //  has what we want still sorted
+    setItems(updatedList);
+    // Update Folder
+    !!folder &&
+      updateFolder({
+        ...folder,
+        content: updatedList.map((item, index) => ({
+          ...item,
+          contentId: item.id,
+          sortOrder: index,
+        })),
+      }).then((data) => {
+        setFolder(data);
+      });
+    // setSortedFlag(!sortedFlag);
+  };
+
+  /** function that will remove items from the folder when the button is clicked */
+  const removeItems = () => {
+    const updatedList = items.filter((item: any) => !selected.includes(item));
+    setItems(updatedList);
+    !!folder &&
+      updateFolder({
+        ...folder,
+        content: updatedList.map((item: any, index: any) => ({
+          ...item,
+          contentId: item.id,
+          sortOrder: index,
+        })),
+      }).then((data) => {
+        setFolder(data);
+        setSortedFlag(!sortedFlag);
+      });
+    setSelected([]);
+  };
+
+  /** determines whether to show body or summary text */
+  const determinePreview = (item: IContentModel) => {
+    if (!!item.body && !item.summary) {
```

Comment:
Why not the following?
`const text = item.body ?? item.summary ?? '';
return parse(text);`

Reply:
Depending on what you're doing here, this is bad logic.
1) Some content types doesn't use the body
2) Some content types doesn't use the summary
3) Transcriptions can only be displayed when they are approved